Sources/Recycling/ListViews/Settings/Cell/SettingsViewToggleCell.swift

```diff
@@ -0,0 +1,75 @@
+//
+//  Copyright © 2019 FINN AS. All rights reserved.
+//
+
+import UIKit
+
+protocol SettingsViewToggleCellDelegate: AnyObject {
+    func settingsViewToggleCell(_ cell: SettingsViewToggleCell, didToggleSettingFor model: SettingsViewToggleCellModel)
+}
+
+class SettingsViewToggleCell: SettingsViewCell {
+
+    // MARK: - Internal Properties
+
+    weak var delegate: SettingsViewToggleCellDelegate?
+
+    // MARK: - Private Properties
+
+    private var model: SettingsViewToggleCellModel?
+
+    private lazy var uiswitch: UISwitch = {
+        let toggle = UISwitch(withAutoLayout: true)
+        toggle.onTintColor = .btnPrimary
+        toggle.addTarget(self, action: #selector(handleValueChanged(toggle:)), for: .valueChanged)
+        return toggle
+    }()
+
+    // MARK: - Init
+
+    override init(style: UITableViewCell.CellStyle, reuseIdentifier: String?) {
+        super.init(style: style, reuseIdentifier: reuseIdentifier)
+        setup()
+    }
+
+    required public init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    // MARK: - Overrides
+
+    override func prepareForReuse() {
+        super.prepareForReuse()
+        configure(with: nil, isLastItem: false)
+    }
+
+    override func setup() {
+        super.setup()
+
+        contentView.addSubview(uiswitch)
+
+        NSLayoutConstraint.activate([
+            uiswitch.centerYAnchor.constraint(equalTo: contentView.centerYAnchor),
+            uiswitch.trailingAnchor.constraint(equalTo: contentView.trailingAnchor, constant: -.mediumLargeSpacing),
+        ])
+    }
+}
+
+extension SettingsViewToggleCell {
+    func configure(with model: SettingsViewToggleCellModel?, isLastItem: Bool) {
+        super.configure(with: model, isLastItem: isLastItem)
+        self.model = model
+        uiswitch.isOn = model?.isOn ?? false
+    }
+}
+
+private extension SettingsViewToggleCell {
+    @objc func handleValueChanged(toggle: UISwitch) {
+        guard var model = model else {
+            return
+        }
+
+        model.isOn = toggle.isOn
```

Comment:
What do you think about making this cell unaware of model and send `isOn` boolean flag from `UISwitch` instead?

Reply:
I quite like this approach. Mutating the model in here would make you sure that the model passed to the delegate is accurate to the actual state of the cell. And it would be related to SwiftUI and how you would use `@State` and `@Binding` in this case.

Reply:
I don't have a strong opinion on that, but the difference between this and SwiftUI is that @State would be a single source of truth. And with our UIKit setup the source of truth is probably an array of settings models in the app, which we use to configure/reload the view from `FinniversKit`. With that in mind I'd make `isOn` property on `SettingsViewToggleCellModel` read-only and make all mutations in the app.

Reply:
You're right, I'll make it read only ;)